framework/src/actions/SplitMeshAction.C

```diff
@@ -59,28 +59,28 @@ SplitMeshAction::act()
 
   // If stripExtension() returns the original string, then there is no
   // file extension or the original string was empty.
-  bool checkpoint_binary_flag = true;
+  bool checkpoint_binary_flag = false;
 
   if (split_file_arg_ext != "")
   {
     if (split_file_arg_ext == "cpr")
       checkpoint_binary_flag = true;
-    else if (split_file_arg_ext == "cpa")
+    else if (split_file_arg_ext == "cpa.gz")
       checkpoint_binary_flag = false;
     else
       mooseError("The argument to --split-file, ",
                  split_file_arg,
-                 ", must not end in a file extension other than .cpr or .cpa");
+                 ", must not end in a file extension other than .cpr or .cpa.gz");
   }
 
   // To name the split files, we start with the given mesh filename
   // (if set) or the argument to --split-file, strip any existing
-  // extension, and then append either .cpr or .cpa depending on the
+  // extension, and then append either .cpr or .cpa.gz depending on the
   // checkpoint_binary_flag.
   auto fname = mesh->getFileName();
   if (fname == "")
     fname = split_file_arg;
-  fname = MooseUtils::stripExtension(fname) + (checkpoint_binary_flag ? ".cpr" : ".cpa");
+  fname = MooseUtils::stripExtension(fname) + (checkpoint_binary_flag ? ".cpr" : ".cpa.gz");
```

Comment:
Same issue here, if we're taking `.cpa.gz` as input too for some reason; `MooseUtils::stripExtension()` also just does `rfind(".")`

Reply:
@roystgnr, are we good to resolve this one? Note that I changed the implementation of `stripExtension` to use find by default.

Reply:
Sorry I missed that; thanks for double-checking.